packaging/flavorgen/cmd/root.go

```diff
@@ -69,13 +134,12 @@ func RunRoot(command *cobra.Command) error {
 			},
 		}
 		util.Replacements = append(util.Replacements, additionalReplacements...)
```

Comment:
Wondering if this could byte us.

E.g. when generating other flavors afterwards we will also have the above replacement active. But seems like it does not make any difference in the output files currently?

Reply:
Just made a little trick on the `util.Replacement` so the order of output files wouldn't become an issue.

Reply:
Jep, that seems to do the trick.

I may have preferred adjusting `generateSingle` to directly return the generated manifests (by calling `util.GenerateManifestYaml`) and passing the replacements array as variable to `GenerateManifestYaml` from there. This way we would have a single place for the overwrites and not a 3rd place for adjustments.

E.g.
```
		return util.GenerateManifestYaml(flavors.ClusterClassTemplateWithKubeVIP(), util.DefaultReplacements), nil
```

Or in the special case:
```
		return util.GenerateManifestYaml(flavors.ClusterTopologyTemplateKubeVIP(), append(util.Replacements, util.Replacement{
			Kind:      "Cluster",
			Name:      "${CLUSTER_NAME}",
			Value:     env.ControlPlaneMachineCountVar,
			FieldPath: []string{"spec", "topology", "controlPlane", "replicas"},
		})), nil
```

But I'm ok with this approach too.

Reply:
I was going to preserve the return value of runtime.Object.
Agreed that returning a string is better.